### PR TITLE
[7.9.x] chore: log agent logs before terminating the container

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
Backport of #299

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests for the CLI, and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Related issues

- Backports #299  

## Follow-ups
When merging this PR, let's not squash the two commits, in the case they backport the new structure for the agent and we have to use the first commit, only

### CODE REVIEW

File: cli/docker/docker.go [hunk 1]
```diff
@@ -102,6 +102,19 @@ func ExecCommandIntoContainer(ctx context.Context, containerName string, user st
 
 	output = strings.ReplaceAll(output, "\n", "")
 
+	patterns := []string{
```
Comment (kuisathaverat) [inline cli/docker/docker.go:105]: my binary is a little rust, What is it for?

Comment (kuisathaverat) [reply cli/docker/docker.go:105]: ok, I saw it https://github.com/elastic/e2e-testing/pull/299/files#r489219873

Comment (kuisathaverat) [inline cli/docker/docker.go:105]:
```suggestion
        // Docker command garbage cleaner :D 
	patterns := []string{
```

Comment (mdelapenya) [reply cli/docker/docker.go:105]: I guess the docker client is returning something weird when converting the bytes. Maybe I'm doing something weird stuff with the buffer and the reader. I'd need to debug it.

Comment (mdelapenya) [reply cli/docker/docker.go:105]: There is a trace log https://github.com/elastic/e2e-testing/pull/310/files#diff-0d15ecc74d96858feb62aab759545abeR114

Should we add the comment too?

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**